[small] test clean up

H-Huang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #128079

remove unnecessary line: https://github.com/pytorch/pytorch/issues/123733
add main so test can be run `python ...`: https://github.com/pytorch/pytorch/issues/124906

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

H-Huang committed:
[small] test clean up

[ghstack-poisoned]

awgu reviewed (approved):
Sounds good if CI is green!

clee2000 commented on test/distributed/elastic/multiprocessing/redirects_test.py:
```diff
@@ -138,3 +138,7 @@ def c_print(i):
+
+
+if __name__ == "__main__":
+    unittest.main()
```
I think you also need to either change the test name to be test_redirects instead of redirects_test or add it to the list of additional tests, see https://github.com/pytorch/pytorch/issues/124317

clee2000 commented on test/run_test.py:
```diff
@@ -745,7 +745,6 @@ def test_distributed(test_module, test_directory, options):
             os.environ["TEMP_DIR"] = tmp_dir
             os.environ["BACKEND"] = backend
-            os.environ["INIT_METHOD"] = "env://"
             os.environ.update(env_vars)
```
Do the lines after this that set the INIT_METHOD env var have any relevance either?  In https://github.com/pytorch/pytorch/issues/123733 it looks like the env var is reset to a different value during setup that doesn't care about this one

H-Huang replied:
Not sure tbh, some others tests might not be using `initialize_temp_directories`. But I'll try removing these lines and see if anything breaks :)

H-Huang committed:
Update on "[small] test clean up"


remove unnecessary line: https://github.com/pytorch/pytorch/issues/123733
add main so test can be run `python ...`: https://github.com/pytorch/pytorch/issues/124906

cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

H-Huang committed:
Update on "[small] test clean up"


remove unnecessary line: https://github.com/pytorch/pytorch/issues/123733
add main so test can be run `python ...`: https://github.com/pytorch/pytorch/issues/124906

cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

H-Huang commented:
@pytorchbot merge
